src/Spark.Engine/Service/FhirServiceExtensions/SearchService.cs

```diff
@@ -182,6 +182,23 @@ private Snapshot CreateSnapshot(Uri selflink, IEnumerable<string> keys, SearchPa
                 selflink = selflink.AddParam(SearchParams.SEARCH_PARAM_REVINCLUDE, searchCommand.RevInclude.Select(inc => inc.Item1).ToArray());
             }
 
+            // add mandatory and modifier elements
+            if (searchCommand.Elements != null && searchCommand.Elements.Any())
+            {
+                var classMapping = _fhirModel.FindClassMapping(type);
```

Comment:
Corresponding to the comment on the IFhirModel interface above, let's do this instead for now:
```csharp
var classMapping = ModelInfo.ModelInspector.FindClassMapping(resourceType);
```